src/Database/Statement/MysqlStatement.php

```diff
@@ -32,6 +32,7 @@ class MysqlStatement extends PDOStatement
      */
     public function execute(?array $params = null): bool
     {
+        $this->_hasExecuted = true;
```

Comment:
I tried extending the unit tests for this statement, but these recursively nested statements makes it very difficult.